pkg/kubernetes/pod/v1alpha1/pod.go

```diff
@@ -142,3 +142,32 @@ func IsNil() Predicate {
 func (p *Pod) GetAPIObject() *corev1.Pod {
 	return p.object
 }
+
+// FromList created a PodList with provided api podlist
+func FromList(pods *corev1.PodList, predicates ...Predicate) *PodList {
```

Comment:
The default `pod.List()` does not have any filters. I wanted `FromList` to have the predicates to filters also else it would be same as `ListBuilderForAPIList`. We can have something like

``schedulednodes = FromList(pods).WithFilters(predicates).GetScheduledNodes()``

But we will be having two `WithFilters` one for builder and one for FromList that is also repeatition I wanted to avoid.

Reply:
For this BDD the predicates are not needed ass I have added two checks before fetching the podList. I have removed the predicates for now but need to think about filtering podList.